### ISSUE TITLE
napi link error on Windows when using crate parcel_css

### ISSUE THREAD

**Issue by dgellow** [parcel-bundler/parcel-css#71]
Hi there,

I'm trying to use the crate `parcel_css` for a rust project on Windows 11 but face a link error related to napi. 

I can reproduce the issue with an empty project importing `parcel_css`.

```toml
# File Cargo.toml
[package]
name = "plop"
version = "0.1.0"
edition = "2021"

[dependencies]
parcel_css = "1.0.0-alpha.13"
```

```rust
// File src/main.rs
use parcel_css::stylesheet::StyleSheet;

fn main() {
    let _style = StyleSheet::parse(
        "plop.css".to_owned(),
        "#hello { color: blue; }",
        Default::default(),
    )
    .unwrap();
}
```

```
$ cargo clean
$ cargo build
 # ...lot of stuff...
  = note: libnapi-41e4dc39fadfbdc1.rlib(napi-41e4dc39fadfbdc1.napi.6ae83a60-cgu.2.rcgu.o) : error LNK2019: unresolved external symbol napi_get_undefined referenced in function _ZN4napi3env3Env13get_undefined17hb7a6765f73f38a65E
 # ... more stuff...
  fatal error LNK1120: 11 unresolved externals
```

---

That being said, I looked a bit into ways to fix the issue and found out my builds would link correctly if I add a build script `build.rs` to my project (the one importing `parcel_css`):

```toml
# File Cargo.toml
# ...

# Add this build dep 👇
[build-dependencies]
napi-build = "1"
```

```rust
// File build.rs
extern crate napi_build;

fn main() {
    napi_build::setup()
}
```

```
$ cargo clean
$ cargo build
   Compiling plop v0.1.0 (C:\Users\Sam\Development\Rust\plop)
warning: cargo:rustc-cdylib-link-arg was specified in the build script of plop v0.1.0 (C:\Users\Sam\Development\Rust\plop), but that package does not contain a cdylib target

Allowing this was an unintended change in the 1.50 release, and may become an error in the future. For more information, see <https://github.com/rust-lang/cargo/issues/9562>.
warning: cargo:rustc-cdylib-link-arg was specified in the build script of plop v0.1.0 (C:\Users\Sam\Development\Rust\plop), but that package does not contain a cdylib target

Allowing this was an unintended change in the 1.50 release, and may become an error in the future. For more information, see <https://github.com/rust-lang/cargo/issues/9562>.
    Finished dev [unoptimized + debuginfo] target(s) in 2.02s
```

(the build yields warnings, but is successful)

--- 

I don't really get why `napi` is required when using the crate `parcel_css`, it seems to be a dependency of `parcel_css_node` but not the other crates.

A few thoughts:
- Should a `build.rs` script be added to `parcel_css` to be sure napi is always built?
- Could the issue be that the cargo workspace is declared in the same Cargo.toml as `parcel_css`?
- It might be a good idea to also run tests on a windows host (via github action).

What do you think?

**Comment by mischnic** [comment]
Napi was probably pulled in by the the parcel sourcemap crate: https://github.com/parcel-bundler/source-map/issues/89#issuecomment-1015090817

**Comment by dgellow** [comment]
I see, thanks @mischnic.

> I think the core crate probably should not depend on napi anyway. 🤔

I agree with this comment from @devongovett in the issue you linked, a napi dependency is problematic here.

**Comment by devongovett** [comment]
Just released parcel_sourcemap v2.0.2, which removes the dependency on napi from the core crate. Let me know if it works for you.

**Comment by dgellow** [comment]
Thanks Devon, it seems to be fixed with `parcel_sourcemap = "2.0.2"` 😸
